lib/chef/resource/aws_internet_gateway.rb

```diff
@@ -42,7 +42,14 @@ class Chef::Resource::AwsInternetGateway < Chef::Provisioning::AWSDriver::AWSRes
 
   def aws_object
     driver, id = get_driver_and_id
-    result = driver.ec2.internet_gateways[id] if id
-    result && result.exists? ? result : nil
+    ec2_resource = ::Aws::EC2::Resource.new(driver.ec2)
+    result = ec2_resource.internet_gateway(id) if id
+    result && exists?(result) ? result : nil
+  end
+
+  def exists?(result)
+    return true if result.data
+  rescue ::Aws::EC2::Errors::InvalidInternetGatewayIDNotFound
```

Comment:
It seems unlikely that `result.data` can raise `::Aws::EC2::Errors::InvalidInternetGatewayIDNotFound` exception. I guess the exception handling block should be in `aws_object` method. Please verify this.

Instead of having a separate method `exists?`, line#47 can be modified as `result && result.data ? result : nil`.